### PR TITLE
Add support for R models

### PR DESCRIPTION
**Pull request by kmariuszk**
This pull request introduces support for generating counterfactuals for models defined in R using the R torch framework. This enhancement expands the capabilities of the package by enabling the utilization of R torch models.

### Changes made

Implemented a basic interface for models defined using R torch.
Created infrastructure to import R torch models with weights stored in a Pickle file.
Modified the methods to ensure compatibility between R torch models and the counterfactual generators.

The R torch interface includes the following components:

- R torch models:
  - Core logic:
    - `src\models\differentiable\R\rtorch_model.jl`
    - `src\generators\gradient_based\loss.jl`
  - Utility logic (model loader function): `src/models/utils.jl`
  - Example: `students_documentation\summary_notebook\summary.qmd`
  - Summary of our efforts to make Python and R models compatible with the generators: `students_documentation\PyTorch_and_R_models_report.md`

Please review the changes and the accompanying documentation.

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -6,3 +6,4 @@ coverage:
         threshold: 0.01%  # the leniency in hitting the target
   ignore:
   - "src/**/visualization/**/*"
+  - "src/models/differentiable/R/rtorch_model.jl"
```
Comment (RaunoArike) [inline codecov.yml:9]: Do you think there's any way to add the loss and loader functions into the ignore section as well? Since they're not testable for now, it shouldn't be something Codecov looks at in my opinion.

**Review by RaunoArike — CHANGES_REQUESTED** [review]
Good job, everything looks good. Requested just one little change.

File: src/models/differentiable/R/rtorch_model.jl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- `logits::AbstractArray`: The probabilities for each output class for the data points in `X`.
+
+# Example
+logits = Models.logits(M, x) # calculates the probabilities for each output class for the data points in `X`
+"""
+function probs(model::RTorchModel, x::AbstractArray)
+    if model.likelihood == :classification_binary
+        return σ.(logits(model, x))
+    elseif model.likelihood == :classification_multi
+        return softmax(logits(model, x))
```
[the remark this answers is not in the document]
Comment (kmariuszk) [reply src/models/differentiable/R/rtorch_model.jl:58]: Good catch, I checked the Python files and I updated it as well.

File: docs/src/tutorials/model_catalogue.qmd [hunk 3]
```diff
@@ -102,6 +102,45 @@ This model can now be passed into the generators like any other.
...
+
+```julia
+model_R = CounterfactualExplanations.Models.RTorchModel(model_loaded, counterfactual_data.likelihood)
+```
+
+### Generating counterfactuals with the R torch model
+
+Now that the R torch model has been wrapped inside the `RTorchModel` class, you can pass it into the counterfactual generators as you would with any other model.
+
+Please note that RCall is not fully compatible with PythonCall. Therefore, it is advisable not to import both R torch and PyTorch models within the same Julia session. Additionally, it's worth mentioning that the R torch integration is still untested in the CounterfactualExplanations package.
```
Comment (RaunoArike) [inline docs/src/tutorials/model_catalogue.qmd:142]: On a second thought, I think we should warn people a bit more saliently that this functionality is untested and might not work together with PythonCall. Maybe we could add a similar warning as the "Breaking changes expected" warning Patrick has [here](https://juliatrustworthyai.github.io/CounterfactualExplanations.jl/stable/tutorials/generators/)?

**Review by RaunoArike — CHANGES_REQUESTED** [review]
I now think that we should be more clear about the fact that this functionality is untested and also provide an overview of why it's impossible to test this in `Python_and_R_models_report.md`. Could you add this?

**Review by VincentPikand — APPROVED** [review]
lgtm

File: docs/src/tutorials/model_catalogue.qmd (hunk 3 above)
Comment (kmariuszk) [reply docs/src/tutorials/model_catalogue.qmd:145]: Very good idea, I've just pushed it.

**Review by RaunoArike — APPROVED** [review]
LGTM, thanks for making the requested changes!
